core/network/protocols/sync_protocol.cpp

```diff
@@ -0,0 +1,342 @@
+
+    log_->info("Read response from outgoing {} stream with {}",
+               protocol_,
+               stream->remotePeerId().value().toBase58());
+
+    read_writer->read<BlocksResponse>(
+        [stream,
+         wp = weak_from_this(),
+         response_handler =
+             std::move(response_handler)](auto &&block_response_res) mutable {
+          auto self = wp.lock();
+          if (not self) {
+            stream->reset();
+            response_handler(Error::GONE);
+            return;
+          }
+
+          if (not block_response_res.has_value()) {
+            self->log_->info(
+                "Error at read response from outgoing {} stream with {}: {}",
+                self->protocol_,
+                stream->remotePeerId().value().toBase58(),
+                block_response_res.error().message());
+
+            stream->reset();
+            response_handler(block_response_res.as_failure());
+            return;
+          }
+          auto &blocks_response = block_response_res.value();
+
+          self->log_->info(
+              "Response read successfuly from outgoing {} stream with {}",
+              self->protocol_,
+              stream->remotePeerId().value().toBase58());
+
+          stream->reset();
+          response_handler(std::move(blocks_response));
+        });
+  }
+
+  void SyncProtocol::request(
+      const PeerId &peer_id,
+      BlocksRequest block_request,
+      std::function<void(outcome::result<BlocksResponse>)> &&response_handler) {
+    auto addresses_res =
+        host_.getPeerRepository().getAddressRepository().getAddresses(peer_id);
+    if (not addresses_res.has_value()) {
+      response_handler(addresses_res.as_failure());
+      return;
+    }
+
+    visit_in_place(
+        block_request.from,
+        [this](primitives::BlockNumber from) {
+          log_->debug("Requesting blocks: from {}", from);
+        },
+        [this, &block_request](const primitives::BlockHash &from) {
+          if (not block_request.to) {
+            log_->debug("Requesting blocks: from {}", from.toHex());
+          } else {
+            log_->debug("Requesting blocks: from {}, to {}",
+                        from.toHex(),
+                        block_request.to->toHex());
+          }
+        });
+
+    newOutgoingStream(
+        {peer_id, addresses_res.value()},
+        [wp = weak_from_this(),
+         response_handler = std::move(response_handler),
+         block_request = std::move(block_request)](auto &&stream_res) mutable {
+          if (not stream_res.has_value()) {
+            response_handler(stream_res.as_failure());
+            return;
+          }
+          auto &stream = stream_res.value();
+
+          auto self = wp.lock();
+          if (not self) {
+            stream->reset();
+            response_handler(Error::GONE);
+            return;
+          }
+
+          self->log_->info("Established outgoing {} stream with {}",
```

Comment:
Not sure if this kind of message should be info instead of verbose.

Reply:
You are right